### DISCUSSION TITLE
How to import .graphql files

### DISCUSSION THREAD

**Discussion by mohas** [vercel/turborepo#7411] [Help] [answered]
Hi, using webpack I used to import graphql files directly in my modules:

next.config.js
```js
config.module.rules.push(
            {
                test: /\.graphql/,
                type: 'asset/source',
                generator: {
                    filename: 'static/chunks/[path][name].[hash][ext]',
                },
            },
        )
```

then in my modules I would do:
```js
import rootDef from './root/root.graphql'

const schema = makeExecutableSchema({
    typeDefs: [rootDef],
    resolvers: ...,
})
```

now turbopack does not recognize importing of .graphql files out of the box, how do I approach this problem?

**Comment by klippx** [comment]
I am in a similar situation, I use NextJS and the webpack option we have to load .graphql files

```
  webpack(config, _options) {
    config.module.rules.push({
      test: /\.(graphql|gql)/,
      exclude: /node_modules/,
      loader: 'graphql-tag/loader',
    })
    // ...
```

This loads .gql files as `ASTNode`, but with turbopack this option seems not currently be supported, at least this loader is not in the [supported list](https://nextjs.org/docs/app/api-reference/config/next-config-js/turbopack#supported-loaders)

Here is also a relevant [SO question](https://stackoverflow.com/questions/79277873/how-to-resolve-graphql-files-in-next-js-using-turbopack)

**Comment by anthonyshew** [comment, marked as answer]
Hi, thanks for the question.

This repository is for Turborepo. For discussions about Turbopack, please post in the Next.js repo.

Thank you!
